src/utils/WethDepositWrapper.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.25;
+
+import "../interfaces/utils/IWethDepositWrapper.sol";
+
+contract WethDepositWrapper is IWethDepositWrapper {
+    using SafeERC20 for IERC20;
+
+    /// @inheritdoc IWethDepositWrapper
+    address public immutable weth;
+    /// @inheritdoc IWethDepositWrapper
+    IVault public immutable vault;
+
+    constructor(IVault vault_, address weth_) {
+        vault = vault_;
+        weth = weth_;
+    }
+
+    /// @inheritdoc IWethDepositWrapper
+    function deposit(
+        address to,
+        address token,
+        uint256 amount,
+        uint256 minLpAmount,
+        uint256 deadline,
```

Comment:
What if check deadline at the first line to reduce gas consumption if it is reached?